src/Operations.jl

```diff
@@ -1210,8 +1210,19 @@ function download_source(ctx::Context, pkgs; readonly = true)
         tracking_registered_version(pkg, ctx.julia_version) || continue
         path = source_path(ctx.env.manifest_file, pkg, ctx.julia_version)
         path === nothing && continue
-        mkpath(dirname(path)) # the `packages/Package` dir needs to exist for the pidfile to be created
-        FileWatching.mkpidlock(() -> ispath(path), path * ".pid", stale_age = pidfile_stale_age) && continue
+        if ispath(path) && iswritable(path)
```

Comment:
Why don't we just do a `ispath(path) && continue`?

Reply:
It used to be like that:

https://github.com/IanButterworth/Pkg.jl/blob/2779d42f4991d09733bf52c9ca47478d1bef3a8d/src/Operations.jl#L1049.

So, why take the pid lock if we won't do anything?

Reply:
Okay, that makes sense to me now.